compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -2851,8 +2851,12 @@ private void initializeBLangVariable(BLangVariable var, TypeDescriptorNode type,
             }
         }
 
+        boolean typeAvailable = false;
+        if (var.getKind() == NodeKind.ERROR_VARIABLE) {
+           typeAvailable = var.typeNode != null;
+        }
```

Comment:
Can't we do

```suggestion
        boolean typeAvailable = var.getKind() == NodeKind.ERROR_VARIABLE && var.typeNode != null;
```

Also wondering if we can have the name reflect that it is an error binding pattern with an error type reference? Like a version of `errorBindingPatternWithTypeRef` maybe?